modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/PartitionMetasInconsistencyOnNodeStartTest.java

```diff
@@ -0,0 +1,217 @@
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** Tests that node with corrupted partition fails on start. */
+    @Test
+    @WithSystemProperty(key = IGNITE_PDS_SKIP_CRC, value = "true")
+    public void testCorruptedNodeFailsOnStart() throws Exception {
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        IgniteCache<Integer, Integer> cache = ignite.getOrCreateCache(new CacheConfiguration<Integer, Integer>(DEFAULT_CACHE_NAME)
+            .setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL)
+            .setAffinity(new RendezvousAffinityFunction(false, 1))
+        );
+
+        cache.put(1, 1);
+
+        PartitionUpdateCounter counter = counter(0, DEFAULT_CACHE_NAME, ignite.name());
+
+        counter.update(10, 5);
+
+        forceCheckpoint();
+
+        final PdsFolderSettings folderSettings = ignite.context().pdsFolderResolver().resolveFolders();
+
+        File storeWorkDir = new File(folderSettings.persistentStoreRootPath(), folderSettings.folderName());
+
+        File cacheWorkDir = new File(storeWorkDir, CACHE_DIR_PREFIX + DEFAULT_CACHE_NAME);
+
+        File partFile = new File(cacheWorkDir, format(PART_FILE_TEMPLATE, 0));
+
+        stopAllGrids();
+
+        FilePageStore store = (FilePageStore)storeFactory.createPageStore(FLAG_DATA, partFile, a -> {});
+
+        store.ensure();
+
+        long metaPageId = findMetaPage(0, FLAG_DATA, store);
+
+        if (metaPageId < 0)
+            fail("Could not find meta page.");
+
+        int pageIdx = PageIdUtils.pageIndex(metaPageId);
+
+        long offset = store.headerSize() + pageIdx * PAGE_SIZE + PART_META_REUSE_LIST_ROOT_OFF;
+
+        writeLongToFileByOffset(store.getFileAbsolutePath(), offset, 0L);
+
+        try {
+            startGrid(0);
+        }
+        catch (Exception ignored) {
+            /* No-op. */
+        }
+
+        assertTrue(correctFailure);
+    }
+
+    /**
+     * Write any number to any place in the file.
+     *
+     * @param path Path to the file.
+     * @param offset Offset.
+     * @param val Value.
+     * @throws Exception If failed.
+     */
+    private void writeLongToFileByOffset(String path, long offset, long val) throws Exception {
+        ByteBuffer buf = ByteBuffer.allocate(8);
+
+        buf.putLong(val);
+
+        buf.rewind();
+
+        try (FileChannel c = new RandomAccessFile(new File(path), "rw").getChannel()) {
+            c.position(offset);
+
+            c.write(buf);
+        }
+    }
+
+    /**
+     * Finds meta page in file page store
+     *
+     * @param partId Partition id.
+     * @param flag Page store flag.
+     * @param store File page store.
+     * @return Page id.
+     * @throws IgniteCheckedException If failed.
+     */
+    protected long findMetaPage(int partId, byte flag, FilePageStore store)
```

Comment:
Maybe _PageMemoryImpl#partitionMetaPageId_ can be used like in other tests? If not, let's move it to some utility class for further use.